redash/query_runner/treasuredata.py

```diff
@@ -102,22 +103,24 @@ def run_query(self, query, user):
                 db=self.configuration.get('db'))
 
         cursor = connection.cursor()
-
-        cursor.execute(query)
-        columns_data = [(row[0], cursor.show_job()['hive_result_schema'][i][1]) for i,row in enumerate(cursor.description)]
-
-        columns = [{'name': col[0],
-            'friendly_name': col[0],
-            'type': TD_TYPES_MAPPING.get(col[1], None)} for col in columns_data]
-
-        if cursor.rowcount == 0:
-            rows = []
-        else:
-            rows = [dict(zip(([c[0] for c in columns_data]), r)) for i, r in enumerate(cursor.fetchall())]
-        data = {'columns': columns, 'rows': rows}
-        json_data = json.dumps(data, cls=JSONEncoder)
-        error = None
-
+        try:
+            cursor.execute(query)
+            columns_data = [(row[0], cursor.show_job()['hive_result_schema'][i][1]) for i,row in enumerate(cursor.description)]
+
+            columns = [{'name': col[0],
+                'friendly_name': col[0],
+                'type': TD_TYPES_MAPPING.get(col[1], None)} for col in columns_data]
+
+            if cursor.rowcount == 0:
+                rows = []
+            else:
+                rows = [dict(zip(([c[0] for c in columns_data]), r)) for i, r in enumerate(cursor.fetchall())]
+            data = {'columns': columns, 'rows': rows}
+            json_data = json.dumps(data, cls=JSONEncoder)
+            error = None
+        except errors.InternalError as e:
+            json_data = None
+            error = "%s: %s" % (e.message, cursor.show_job()['debug']['stderr'])
```

Comment:
Is `['debug']['stderr']` guaranteed to be always present? Maybe worth safe guarding with `.get()`?

Reply:
Thank you for your review.
I think it's guaranteed. But, yeah, I agree with you.
I'll add .get() for it.